compiler/noirc_frontend/src/hir/def_collector/dc_mod.rs

```diff
@@ -986,6 +986,22 @@ mod find_module_tests {
         find_module(&fm, lib_file_id, "foo").unwrap();
     }
 
+    #[test]
+    fn errors_mod_dot_nr_in_same_directory() {
```

Comment:
As we're always checking `mod.nr` as a fallback, this leads `lib.nr` with a `mod.nr` at the same directory level resolving all modules which aren't defined to this `mod.nr` file. (This would also result in a `mod.nr` file being able to define as many modules as it wants which will all resolve back to itself.

We should take care that a file with `mod bar` can only look at the paths:
- `lib.nr` -> `./bar.nr` and `./bar/mod.nr`
- `foo.nr` -> `./foo/bar.nr` and `./foo/bar/mod.nr`